### PR TITLE
Create Swagger based REST API

### PR DESCRIPTION
**Pull request by metalmatze**
This creates a first Swagger definition for the GitPods API. I don't want to add too many paths just yet, to make the overall direction as obvious as possible.

Take a look at the first working draft for a [swagger.yaml](https://github.com/gitpods/gitpods/pull/22/files#diff-bac86ae86657c2fa2c3dd2ab26df4960).
In [cmd/api/api.go](https://github.com/gitpods/gitpods/pull/22/files#diff-629e64d5bc053b1aa391feaaa4704990) we simply create the `api/v1` and mount its Handler to the already existing router. For now it will serve behind the same middleware that authenticates via cookies, but that's going to change eventually.
The most interesting files is [internal/api/v1/api.go](https://github.com/gitpods/gitpods/pull/22/files#diff-d5527cacb9976050ed370728596fd056) that takes the already existing `user.Service` as interface to create a new API. Next we register the Swagger handler with 3 HandlerFuncs that I've written myself to convert Swagger to the internal interfaces. Pretty straight forward now and we can generate all files in `internal/api/v1/{models,restapi}` as often as we want, because we pass in our functions. :+1: 

Let me know in Discord if you have more questions.

/cc @tboerger @bkcsoft @lafriks @cbrgm

### CODE REVIEW

File: cmd/api/api.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 
 	"github.com/gitpods/gitpods/authorization"
 	"github.com/gitpods/gitpods/cmd"
+	apiv1 "github.com/gitpods/gitpods/internal/api/v1"
```
Comment (bkcsoft) [inline cmd/api/api.go]: I would just import this as `api "..."` (we could change it if we get to v2)

Comment (tboerger) [reply cmd/api/api.go]: IMHO it's a good convention to directly clearly define it as v1

File: internal/api/v1/api.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+				})
+			}
+			return users.NewGetUserDefault(http.StatusInternalServerError)
+		}
+
+		return users.NewGetUserOK().WithPayload(convertUser(u))
+	}
+}
+
+func UpdateUserHandlerFunc(us user.Service) users.UpdateUserHandlerFunc {
```
Comment (tboerger) [inline internal/api/v1/api.go]: The other handler implementations don't have the Func suffix

File: cmd/api/api.go (hunk 1 above)
Comment (metalmatze) [reply cmd/api/api.go]: As this is just the alias I don't have a strong opinion on this, but I think that the apiv1 alias is already the future proof solution. I'll simply leave it like this.

File: internal/api/v1/restapi/operations/users/list_users_parameters.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	// HTTP Request Object
+	HTTPRequest *http.Request `json:"-"`
+}
+
+// BindRequest both binds and validates a request, it assumes that complex things implement a Validatable(strfmt.Registry) error interface
+// for simple values it will use straight method calls.
+//
+// To ensure default values, the struct must have been initialized with NewListUsersParams() beforehand.
+func (o *ListUsersParams) BindRequest(r *http.Request, route *middleware.MatchedRoute) error {
```
[the remark this answers is not in the document]
Comment (fatih) [reply internal/api/v1/restapi/operations/users/list_users_parameters.go]: Interesting, this seems to be a bug. I'm going to check it :)

[the remark this answers is not in the document]
Comment (metalmatze) [reply internal/api/v1/restapi/operations/users/list_users_parameters.go]: Ha! Indeed, good catch!

[the remark this answers is not in the document]
Comment (fatih) [reply internal/api/v1/restapi/operations/users/list_users_parameters.go]: This is now fixed, new PR's won't have this problem anymore.

**Pull request merged**